cypher-shell/src/main/java/org/neo4j/shell/state/BoltStateHandler.java

```diff
@@ -133,7 +133,20 @@ private void silentDisconnect() {
         }
     }
 
+    /**
+     * Reset the current session. This rolls back any open transactions.
+     */
     public void reset() {
-//        session.reset();
+        // TODO once drivers release next milestone
+        //if (session != null) {
+            // session.reset();
+        //}
+
+        // Clear current state
```

Comment:
I'm fine with us resetting `tx` to null, but I wonder what happens when you close an already closed transaction etc.. If this is tested and we are ok with it, I am alright :)